incubator/hnc/internal/reconcilers/anchor.go

```diff
@@ -279,12 +279,17 @@ func (r *AnchorReconciler) deleteNamespace(ctx context.Context, log logr.Logger,
 	return nil
 }
 
-func (r *AnchorReconciler) allowsCascadingDelete(nm string) bool {
+// canDelete returns true if the namespace still exists and it is a leaf
+// subnamespace or it allows cascading delete.
+func (r *AnchorReconciler) canDelete(nm string) bool {
 	r.forest.Lock()
 	defer r.forest.Unlock()
 
 	ns := r.forest.Get(nm)
-	return ns.AllowsCascadingDelete()
+	if !ns.Exists() {
```

Comment:
This seems like an odd place for this line. Why didn't we need it before, and why do we need it now?